controller-server/src/main/java/com/yahoo/vespa/hosted/controller/auditlog/AuditLogger.java

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.controller.auditlog;
+
+import com.yahoo.container.jdisc.HttpRequest;
+import com.yahoo.vespa.curator.Lock;
+import com.yahoo.vespa.hosted.controller.persistence.CuratorDb;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.security.Principal;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This provides read and write operations for the audit log.
+ *
+ * @author mpolden
+ */
+public class AuditLogger {
+
+    /** The TTL of log entries. Entries older than this will be removed when the log is updated */
+    private static final Duration entryTtl = Duration.ofDays(14);
+
+    private final CuratorDb db;
+    private final Clock clock;
+
+    public AuditLogger(CuratorDb db, Clock clock) {
+        this.db = Objects.requireNonNull(db, "db must be non-null");
+        this.clock = Objects.requireNonNull(clock, "clock must be non-null");
+    }
+
+    /** Read the current audit log */
+    public AuditLog get() {
```

Comment:
1. I think get() is odd when it returns something else than itself. 
2. I suggest calling this something with read so it is clear that it is not cheap.